api/specs.proto

```diff
@@ -101,6 +102,7 @@ message TaskSpec {
 	oneof runtime {
 		NetworkAttachmentSpec attachment = 8;
 		ContainerSpec container = 1;
+		CustomSpec custom = 10;
```

Comment:
The word "custom" is probably a little narrow. "dynamic" or something in that area would would be more apt. Basically, the goal is to decouple the swarmkit schema from the runtime schema.

Why not just embed the any field here directly and use the url to name the runtime?